pkg/kapp/diff/change_set_with_versioned_rs.go

```diff
@@ -76,6 +84,72 @@ func (d ChangeSetWithVersionedRs) Calculate() ([]Change, error) {
 	return allChanges, nil
 }
 
+func (d ChangeSetWithVersionedRs) checkAndUpdateResWithMaxDurationAnn(existingRsGrouped map[string][]ctlres.Resource,
+	existingRs versionedResources) error {
+
+	exNonVersionResMap := map[string]ctlres.Resource{}
+
+	for _, res := range existingRs.NonVersioned {
+		resKey := VersionedResource{res, nil}.UniqVersionedKey().String()
+		exNonVersionResMap[resKey] = res
+	}
+
+	for _, res := range d.newRs {
+		if val, found := res.Annotations()[maxDurationAnnKey]; found {
+			duration, err := time.ParseDuration(val)
+			if err != nil {
+				return fmt.Errorf("Unable to parse max-duration for resource %s: %s", res.Description(), err.Error())
+			}
+
+			resKey := VersionedResource{res, nil}.UniqVersionedKey().String()
+			exRes, found := exNonVersionResMap[resKey]
+			if !found {
+				if exResGroup, found := existingRsGrouped[resKey]; found {
+					exRes = exResGroup[len(exResGroup)-1]
+				}
+			}
+
+			if exRes != nil {
+				err = addLastRenewedTimeAnn(res, exRes, duration)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func addLastRenewedTimeAnn(res, exRes ctlres.Resource, duration time.Duration) error {
+	var (
+		lastRenewed, creationTime time.Time
+		err                       error
+	)
+
+	if val, found := exRes.Annotations()[lastRenewTimeAnnKey]; found {
```

Comment:
Can we simplify it like this(Reason: Our logic is based on lastRenewedTime rather than creationTime:

if val, found := exRes.Annotations()[lastRenewTimeAnnKey]; found {
   ...
} else {
lastRenewed = exRes.CreatedAt()
}

if time.Now().Before(lastRenewed.Add(duration)) {
		return nil
	}
....

Reply:
We want to use the found check to decide which time to use which makes sense.
I would also say we should split up the conditional into two lines (no strong opinions on this one)

Reply:
This will require one more if check inside if case. 
Something like:
```
if val, found := exRes.Annotations()[lastRenewTimeAnnKey]; found {
...
      if lastRenewed.Before(exRes.CreatedAt()) {
		lastRenewed = exRes.CreatedAt()
	}
} else {
lastRenewed = exRes.CreatedAt()
}

if time.Now().Before(lastRenewed.Add(duration)) {
return nil
}
```
Existing implementation makes more sense to me.